library/std/src/Std/InternalHelpers.qs

```diff
@@ -90,6 +83,9 @@ internal operation CT(control : Qubit, target : Qubit) : Unit is Adj {
     CNOT(control, target);
     Adjoint Rz(angle, target);
     CNOT(control, target);
+    // This decomposition for controlled-T introduces a global phase (due to the unmatched call to Rz from above).
+    // We correct for this global phase in simulation with R(PauliI), which is a no-op on hardware.
+    Adjoint R(PauliI, angle, target);
```

Comment:
If the purpose of this is just to apply global phase (rather than provide some canonical decomposition form) maybe just call ApplyGlobalPhase?
ApplyGlobalPhase(angle / 2.0)

Reply:
That's a good point, this is an internal helper so there is no need to worry about expressing it in terms of `R`. I'll make that change.